packages/test/test-end-to-end-tests/src/test/idCompressor.spec.ts

```diff
@@ -0,0 +1,326 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { SharedMap } from "@fluidframework/map";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import {
+	DataObjectFactoryType,
+	ITestContainerConfig,
+	ITestFluidObject,
+	ITestObjectProvider,
+} from "@fluidframework/test-utils";
+import { describeNoCompat } from "@fluid-internal/test-version-utils";
+import { SharedCell } from "@fluidframework/cell";
+import { IIdCompressor, SessionSpaceCompressedId } from "@fluidframework/runtime-definitions";
+import { SharedObjectCore } from "@fluidframework/shared-object-base";
+
+const mapId = "mapKey";
+const cellId = "cellKey";
+const testContainerConfig: ITestContainerConfig = {
+	registry: [
+		[mapId, SharedMap.getFactory()],
+		[cellId, SharedCell.getFactory()],
+	],
+	runtimeOptions: {
+		enableRuntimeIdCompressor: true,
+	},
+	fluidDataObjectType: DataObjectFactoryType.Test,
+};
+
+function getIdCompressor(dds: SharedObjectCore): IIdCompressor {
+	return (dds as any).runtime.idCompressor as IIdCompressor;
+}
+
+describeNoCompat("Runtime IdCompressor", (getTestObjectProvider) => {
+	let provider: ITestObjectProvider;
+	beforeEach(() => {
+		provider = getTestObjectProvider();
+	});
+
+	let sharedMap1: SharedMap;
+	let sharedCell1: SharedCell;
+	let sharedMap2: SharedMap;
+	let sharedMap3: SharedMap;
+
+	beforeEach(async () => {
+		const container1 = await provider.makeTestContainer(testContainerConfig);
+		const dataObject1 = await requestFluidObject<ITestFluidObject>(container1, "default");
+		sharedMap1 = await dataObject1.getSharedObject<SharedMap>(mapId);
+		sharedCell1 = await dataObject1.getSharedObject<SharedCell>(cellId);
+
+		const container2 = await provider.loadTestContainer(testContainerConfig);
+		const dataObject2 = await requestFluidObject<ITestFluidObject>(container2, "default");
+		sharedMap2 = await dataObject2.getSharedObject<SharedMap>(mapId);
```

Comment:
Naming these map's something like `sharedMapContainer1`, `sharedMapContainer2`, `sharedMapContainer3` will help following whats happening in the test much better. For example, some of tests deliberately do some operations on container1's map and others on container2's map to validate cross client collaboration but its not clear. This naming will help in these tests.

Reply:
Nice, that should help with clarity a lot. I've updated the variable names as suggested.